tools/quake3/common/polylib.c

```diff
@@ -68,7 +68,7 @@ winding_t   *AllocWinding( int points ){
 			c_peak_windings = c_active_windings;
 		}
 	}
-	s = sizeof( vec_t ) * 3 * points + sizeof( int );
+	s = sizeof( *w ) + points * sizeof( *w->p );
```

Comment:
Wouldn't this be easier to parse? ```s = offsetof( w->p ) + points * sizeof( *w->p )```

Reply:
That would be the thing I would try to get *away* from: one of the main selling points of flexible arrays is that it allows you to use `sizeof` to get the size of the non-flexible header. Both versions work, of course, but I think the `sizeof` version is "idiomatic", whereas the `offsetof` version has always been a kind of a hack.

Your call, though, they're entirely equivalent. Depending on which way you want it, I may or may not send a follow-up to modify existing `offsetof`s.

Reply:
Makes sense, thanks. I agree with using sizeof whenever possible.

Reply:
@tkoeppe just one question though, on NetRadiant side this line was written this way:

```c
s = sizeof( *w ) + ( points ? sizeof( w->p[0] ) * ( points - 1 ) : 0 );
```

Do you have any idea why someone would have done that `sizeof( w->p[0] ) * ( points - 1 )` trick?

Reply:
@illwieckz: That's presumably because they *don't* use a flexible array, but instead use an array of size 1, which was what people did in C89 (and why C99 added the flexible array support). That way, if you use `sizeof` rather than offsets, you have to compensate for the extraneous one element. That approach is brittle, though, when alignment and padding are a concern.